src/overlays/actors/ovl_En_Nwc/z_en_nwc.c

```diff
@@ -36,47 +54,474 @@ const ActorInit En_Nwc_InitVars = {
+            this->actor.world.rot.y = this->actor.home.rot.z * 0x3000 & 0xFFFF; // Fake Match?: & 0xFFFF
+            this->actor.shape.rot.y = this->actor.world.rot.y;
+            this->stateTimer = Rand_ZeroFloat(40.0f) + 120.0f;
+            this->actionFunc = EnNwc_RunAway;
+            break;
+    }
+
+    this->state = newState;
+}
+
+/**
+ * Summary: Changes the current actor state
+ *     If previously random behavior -> check if breman mask is active
+ *     If previously checking for breman -> select random (NWC_STATE_TURNING, NWC_STATE_HOPPING_FORWARD)
+ */
+void EnNwc_ToggleState(EnNwc* this) {
+    this->actor.speedXZ = 0.0f;
+    if (this->state != NWC_STATE_CHECK_BREMAN) {
+        EnNwc_ChangeState(this, NWC_STATE_CHECK_BREMAN);
+    } else {
+        EnNwc_ChangeState(this, Rand_ZeroFloat(2.0f) + 1.0f);
+    }
+}
+
+void EnNwc_CheckFound(EnNwc* this, GlobalContext* globalCtx) {
+    if (EnNwc_IsFound(this, globalCtx)) {
+        u8 currentChickCount = (this->grog->actor.home.rot.z / 2);
+
+        if (currentChickCount > 9) {
+            currentChickCount = 9;
+        }
+
+        // save our current chick order
+        this->actor.home.rot.z = this->grog->actor.home.rot.z + 1;
+
+        // if < 10 chicks, increment grog's chick counter
+        if (this->grog->actor.home.rot.z < 20) {
+            this->grog->actor.home.rot.z += 2;
+        }
 
-extern UNK_TYPE D_060000E8;
-extern UNK_TYPE D_060002E8;
+        EnNwc_ChangeState(this, NWC_STATE_FOLLOWING);
+        func_801A0868(&D_801DB4A4, NA_SE_SY_CHICK_JOIN_CHIME, currentChickCount);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Nwc/EnNwc_Init.s")
+void EnNwc_LoadNiwSkeleton(EnNwc* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->niwObjectIndex)) {
+        gSegments[6] = PHYSICAL_TO_VIRTUAL(globalCtx->objectCtx.status[this->niwObjectIndex].segment);
+
+        SkelAnime_InitFlex(globalCtx, &this->niwSkeleton, &gNiwSkeleton, &gNiwIdleAnim, this->jointTable,
+                           this->morphTable, NIW_LIMB_MAX);
+        Animation_Change(&this->niwSkeleton, &gNiwIdleAnim, 1.0f, 0.0f, Animation_GetLastFrame(&gNiwIdleAnim), 0, 0.0f);
+
+        gSegments[6] = PHYSICAL_TO_VIRTUAL(globalCtx->objectCtx.status[this->nwcObjectIndex].segment);
+        this->state = NWC_STATE_NIW_LOADED;
+        EnNwc_ToggleState(this);
+    }
+}
+
+void EnNwc_CrowAtTheEnd(EnNwc* this, GlobalContext* globalCtx) {
+    // I guess grog handles the scene transit?
+    Math_SmoothStepToS(&this->upperBodyRotY, 0x2710, 2, 0x1B58, 0x3E8);
+    Math_SmoothStepToS(&this->footRotZ, 0, 2, 0x1B58, 0x3E8);
+    Math_SmoothStepToS(&this->footRotY, 0x36B0, 2, 0x1B58, 0x3E8);
+}
+
+/**
+ * Summary: Controls the Chick when following Breman Mask.
+ *
+ * ActionFunc for NWC Type: NWC_STATE_FOLLOWING
+ */
+void EnNwc_Follow(EnNwc* this, GlobalContext* globalCtx) {
+    Vec3f* chickCoords = ((EnHs*)this->grog)->nwcPos;
```

Comment:
```suggestion
    Vec3f* chickCoords = this->grog->nwcPos;
```
No need for the cast anymore.